core/src/main/java/be/orbinson/aem/dictionarytranslator/utils/DictionaryUtil.java

```diff
@@ -0,0 +1,30 @@
+package be.orbinson.aem.dictionarytranslator.utils;
+
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.Nullable;
+
+import static com.day.cq.commons.jcr.JcrConstants.JCR_LANGUAGE;
+
+public class DictionaryUtil {
+
+    private DictionaryUtil() {
+    }
+
+    /**
+     * Gets the language resource based on the jcr:language property
+     *
+     * @param dictionaryResource The dictionary resource
+     * @param language           The language
+     * @return the language resource if it exists
+     */
+    public static @Nullable Resource getLanguageResource(Resource dictionaryResource, String language) {
```

Comment:
I would rather put this into `DictionaryService` and clarify in the javadoc the difference to `getLanguages(Resource)`

Reply:
Maybe one can even combine both and let `getLanguages(Resource)` return a `Map<Locale,String>` (i.e. key = language and value = path/resource). Not sure how easy both are to pass to the servlets, though.